curator-client/src/main/java/org/apache/curator/ensemble/exhibitor/ExhibitorEnsembleProvider.java

```diff
@@ -301,7 +301,7 @@ else if ( port != thePort )
                     values.putAll(decodeExhibitorList(encoded));
                     done = true;
                 }
-                catch ( Throwable e )
+                catch ( Exception e )
```

Comment:
You also need to \worry about ThreadDeath error:

https://docs.oracle.com/javase/7/docs/api/java/lang/ThreadDeath.html
```
An application should catch instances of this class only if it must clean up after being terminated asynchronously. If ThreadDeath is caught by a method, it is important that it be rethrown so that the thread actually dies.
```

So ThreadDeath Error can be logged, but should never retry to avoid destabilizing JVM.